django/core/files/storage/filesystem.py

```diff
@@ -98,12 +130,22 @@ def _save(self, name, content):
             try:
                 # This file has a file path that we can move.
                 if hasattr(content, "temporary_file_path"):
-                    file_move_safe(content.temporary_file_path(), full_path)
+                    file_move_safe(
+                        content.temporary_file_path(),
+                        full_path,
+                        allow_overwrite=self._allow_overwrite,
+                    )
 
                 # This is a normal uploadedfile that we can stream.
                 else:
-                    # The current umask value is masked out by os.open!
-                    fd = os.open(full_path, self.OS_OPEN_FLAGS, 0o666)
+                    if (
+                        self._default_open_flags == self.OS_OPEN_FLAGS
+                        and self._allow_overwrite
+                    ):
+                        open_flags = self._default_open_flags & ~os.O_EXCL
+                    else:
+                        open_flags = self.OS_OPEN_FLAGS
+                    fd = os.open(full_path, open_flags, 0o666)
```

Comment:
This is missing a comment as to how this will simplify after the OS_OPEN_FLAGS is removed.
If you search for `# RemovedInDjango60Warning: when the deprecation ends, replace with:` in the codebase I think you'll get what I mean 👍

Reply:
I added comments about the deprecation end.